client/src/components/AnchorSelectorField/tests/AnchorSelectorField-test.js

```diff
@@ -1,103 +1,110 @@
+});
 
-  describe('getDropdownOptions()', () => {
-    it('Merges value with page anchors', () => {
-      field = ReactTestUtils.renderIntoDocument(<AnchorSelectorField {...props} />);
-      expect(field.getDropdownOptions()).toEqual([
-        { value: 'selectedanchor' },
-        { value: 'anchor1' },
-        { value: 'anchor2' },
-      ]);
-    });
-  });
+test('AnchorSelectorField componentDidMount() Does not load success selectors', async () => {
+  const beginUpdating = jest.fn();
+  render(<AnchorSelectorField {...makeProps({
+    loadingState: anchorSelectorStates.SUCCESS,
+    actions: {
+      anchorSelector: {
+        beginUpdating,
+        updated: () => {},
+        updateFailed: () => {},
+      },
+    },
+  })}
+  />);
+  await new Promise(resolve => setTimeout(resolve, 0));
+  expect(beginUpdating).not.toBeCalled();
+});
 
-  describe('ensurePagesLoaded', () => {
-    it('Triggers loading on dirty', () => {
-      props.loadingState = anchorSelectorStates.DIRTY;
-      field = ReactTestUtils.renderIntoDocument(<AnchorSelectorField {...props} />);
-      return field
-        .ensurePagesLoaded()
-        .then((result) => {
-          expect(props.actions.anchorSelector.beginUpdating)
-            .toHaveBeenCalledWith(4);
-          expect(props.actions.anchorSelector.updated)
-            .toHaveBeenCalledWith(4, ['anchor3', 'anchor4']);
-          expect(props.actions.anchorSelector.updateFailed)
-            .not
-            .toHaveBeenCalled();
-          expect(result).toEqual(['anchor3', 'anchor4']);
-        });
-    });
+test('AnchorSelectorField ensurePagesLoaded Triggers loading on dirty', async () => {
+  const beginUpdating = jest.fn();
+  const updated = jest.fn();
+  const updateFailed = jest.fn();
+  render(<AnchorSelectorField {...makeProps({
+    loadingState: anchorSelectorStates.DIRTY,
+    actions: {
+      anchorSelector: {
+        beginUpdating,
+        updated,
+        updateFailed,
+      },
+    },
+  })}
+  />);
+  await new Promise(resolve => setTimeout(resolve, 0));
+  expect(beginUpdating).toBeCalledWith(4);
+  expect(updated).toBeCalledWith(4, ['anchor3', 'anchor4']);
+  expect(updateFailed).not.toBeCalled();
+});
 
-    it('Does not trigger updating', () => {
-      props.loadingState = anchorSelectorStates.UPDATING;
-      field = ReactTestUtils.renderIntoDocument(<AnchorSelectorField {...props} />);
-      return field
-        .ensurePagesLoaded()
-        .then((result) => {
-          expect(props.actions.anchorSelector.beginUpdating)
-            .not
-            .toHaveBeenCalled();
-          expect(props.actions.anchorSelector.updated)
-            .not
-            .toHaveBeenCalled();
-          expect(props.actions.anchorSelector.updateFailed)
-            .not
-            .toHaveBeenCalled();
-          expect(result).toBe(undefined);
-        });
-    });
-  });
+test('AnchorSelectorField ensurePagesLoaded Does not trigger updating', async () => {
```

Comment:
The `expect(result).toBe(undefined);` part of this test is missing. As above, it's not clear what that was testing exactly but we should probably either explicitly state what it was and why it _shouldn't_ be tested, or find a way to test it.

Reply:
`expect(result).toBe(undefined);` was essentially testing the return value of the class method `ensurePagesLoaded()

We're not testing the class method directly anymore

Reply:
> We're not testing the class method directly anymore

That's my point exactly. I haven't looked at the source code but I assume that method is called within the component, and its return value is used somewhere for some purpose, and therefore that value is important.

If that's not the case then fine, but if that is the case then we need to find some way to test that this is returning the correct value - or that the end result of whatever the value is used for is correct.

Reply:
Upon further discussion offline, it has been decided that there would likely be more effort required to either validate this or implement a way to continue testing what was being tested than it's worth.